File: euler/03/3.cpp
#include <cmath>
#include <iostream>

int main(){

	long long num = 600851475143;

	printf("The number is %lld\n", num);
	
	const long long upper = sqrt(num);
	long max = 0;

	for(long i=2; i<upper; i++){
		if(num % i==0){
			max = i;
			while(num % i ==0){
				num /= i;
			}
		}
	}
	
	printf("The largest prime divisor of %lld is %ld\n",num,max); 

}

File: euler/03/3.f90
program euler3
    implicit none

    integer, parameter :: sk = selected_int_kind(12)
    integer(kind=sk), parameter :: num=600851475143_sk
    integer(kind=sk), parameter :: upper = sqrt(num*1.0)+1
    integer(kind=sk) :: cur_max=-1, lnum=num
    integer :: i 

    do i=2,upper
        if(mod(lnum, i) == 0) then
            cur_max = i
            do while(mod(lnum, i) == 0)
                lnum = lnum/i
            enddo
        endif
    enddo 

    write(*,*) "The largest prime divisor of ", num, "is", cur_max

end program euler3
